simulations/src/streaming/naive.rs

```diff
@@ -102,8 +113,34 @@ where
 
     fn sink(&self, state: Arc<Self::Record>) -> anyhow::Result<()> {
         let mut file = self.file.lock();
-        serde_json::to_writer(&mut *file, &state)?;
-        file.write_all(b",\n")?;
+        match self.format {
+            SubscriberFormat::Json => {
+                serde_json::to_writer(&mut *file, &state)?;
+            }
+            SubscriberFormat::Csv => {
+                let mut w = csv::Writer::from_writer(&mut *file);
```

Comment:
When using `w.serialize()` with `serde`, writer by default adds headers on every `w.flush()`. This makes csv headers appear multiple times in this case. 
We could use `WriterBuilder` to avoid that:
```
let mut w = csv::WriterBuilder::new()
    .has_headers(false)
    .from_writer(&mut *file);
```

Reply:
Good catch! Thanks!